packages/flutter_tools/lib/src/doctor.dart

```diff
@@ -95,34 +95,53 @@ class Doctor {
     return buffer.toString();
   }
 
-  /// Print verbose information about the state of installed tooling.
-  Future<bool> diagnose({ bool androidLicenses: false }) async {
+  /// Print information about the state of installed tooling.
+  Future<bool> diagnose({ bool androidLicenses: false, bool verbose: true }) async {
     if (androidLicenses)
       return AndroidWorkflow.runLicenseManager();
 
     bool doctorResult = true;
+    int issues = 0;
 
     for (DoctorValidator validator in validators) {
       final ValidationResult result = await validator.validate();
 
-      if (result.type == ValidationType.missing)
+      if (result.type == ValidationType.missing ) {
         doctorResult = false;
+      }
+      if (result.type != ValidationType.installed) {
+        issues += 1;
+      }
 
       if (result.statusInfo != null)
         printStatus('${result.leadingBox} ${validator.title} (${result.statusInfo})');
       else
         printStatus('${result.leadingBox} ${validator.title}');
 
       for (ValidationMessage message in result.messages) {
-        final String text = message.message.replaceAll('\n', '\n      ');
-        if (message.isError) {
-          printStatus('    ✗ $text', emphasis: true);
-        } else {
-          printStatus('    • $text');
+        if (message.isError || message.isHint || verbose == true) {
+          final String text = message.message.replaceAll('\n', '\n      ');
+          if (message.isError) {
+            printStatus('    ✗ $text', emphasis: true);
+          } else if (message.isHint) {
+            printStatus('    ! $text');
+          } else {
+            printStatus('    • $text');
+          }
         }
       }
+      if (verbose)
+        printStatus('');
+    }
 
+    // Make sure there's always one line before the summary even when not verbose.
+    if (!verbose)
       printStatus('');
+    if (issues > 0) {
+      final String verboseSuggestion = verbose ? '.' : "; run 'flutter doctor --verbose' for more details.";
```

Comment:
--verbose doesn't actually give more details about the issues (which is good). The way this is phrased makes it sound like it does.

Reply:
done.